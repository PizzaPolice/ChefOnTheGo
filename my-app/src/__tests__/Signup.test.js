import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router';
import { shallow, mount, render } from 'enzyme';
import Signup from '../components/signup.js'


jest.mock('../App');

/*
it('Signup component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Signup />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
*/

// This test is shallow rendering our Signup component
describe('Signup Component', () => {
    it('Signup should render without throwing an error', () => {
        expect(shallow(<Signup />).exists(<form className='Signup'></form>)).toBe(true)
      })
      it("should render initial layout", () => {
        const component = shallow(<Signup />);
        expect(component.getElements()).toMatchSnapshot();
        })
    });
