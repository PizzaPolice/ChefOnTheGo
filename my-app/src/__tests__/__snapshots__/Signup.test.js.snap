// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component should render initial layout 1`] = `
Array [
  <div
    className="outer-panel"
>
    <div
        className="panel-body"
    >
        <h2
            className="title"
        >
            Sign up
        </h2>
        <div
            className="input-fields"
        >
            <div
                className="input-line"
            >
                Username:
                <input
                    name="username"
                    onChange={[Function]}
                    type="text"
                />
            </div>
            <div
                className="input-line"
            >
                Password:
                <input
                    name="password"
                    onChange={[Function]}
                    type="password"
                />
            </div>
            <div
                className="input-line"
            >
                Re-enter Password:
                <input
                    name="check_password"
                    onChange={[Function]}
                    type="password"
                />
            </div>
            <div
                className="input-line"
            >
                Email:
                <input
                    name="email"
                    onChange={[Function]}
                    type="text"
                />
            </div>
            <div
                className="radio-input"
            >
                <label
                    htmlFor="customer"
                >
                    Customer
                </label>
                <input
                    id="customer"
                    name="login_type"
                    onChange={[Function]}
                    type="radio"
                    value="customer"
                />
                <label
                    htmlFor="chef"
                >
                    Chef
                </label>
                <input
                    id="chef"
                    name="login_type"
                    onChange={[Function]}
                    type="radio"
                    value="chef"
                />
            </div>
        </div>
    </div>
    <div
        className="panel-footer"
    >
        <Link
            onClick={[Function]}
            replace={false}
            to="login"
        >
            <button
                className="btn next-btn"
            >
                Sign up
            </button>
        </Link>
    </div>
</div>,
]
`;
